docs/src/man/comparisons.md

```diff
@@ -1,77 +1,191 @@
 # Comparisons
 
-This section compares DataFrames.jl with other data manipulation frameworks.
+This section compares DataFrames.jl with other data manipulation frameworks in Python, R, and Stata.
+
+A sample data set can be created using the following code:
+
+```julia
+df = DataFrame(id = 'a':'f', grp = repeat(1:2, 3), x = 6:-1:1, y = 4:9, z = [3:7; missing])
+df2 = DataFrame(grp = 1, w = 10)
+```
+
+## Comparison with the Python package pandas
+
+The following table compares the main functions of DataFrames.jl with the Python package pandas (version 1.1.0):
+
+```python
+>>> df = pandas.DataFrame({'grp': [1, 2, 1, 2, 1, 2],
+...                        'x': range(6, 0, -1),
+...                        'y': range(4, 10),
+...                        'z': [3, 4, 5, 6, 7, None]},
+...                       index = list('abcdef'))
+>>> df
+   grp  x  y    z
+a    1  6  4  3.0
+b    2  5  5  4.0
+c    1  4  6  5.0
+d    2  3  7  6.0
+e    1  2  8  7.0
+f    2  1  9  NaN
+```
+
+By comparison, this pandas data frame has `a` to `f` as row indices rather than a separate `id` column.
+
+### Accessing data
+
+| Operation                  | pandas                 | DataFrames.jl            |
+|:---------------------------|:-----------------------|:-------------------------|
+| Cell indexing by location  | `df.iloc[1, 1]`        | `df[2, 2]`               |
+| Row slicing by location    | `df.iloc[1:3]`         | `df[2:3, :]`             |
+| Column slicing by location | `df.iloc[:, 1:]`       | `df[:, 2:end]`           |
+| Row indexing by label      | `df.loc['c']`          | `df[df.id .== 'c', :]`   |
```

Comment:
I think this is the equivalent:
```suggestion
| Row indexing by label      | `df.loc[['c']]`          | `df[df.id .== 'c', :]`   |
```

Reply:
Actually checking this I have realized that pandas allows duplicate keys and finds only the first one - this is something that we would really not like to introduce in DataFrames.jl as at least I feel it is unsafe. I much prefer our `groupby` approach.

Reply:
The proposed syntax is more performant, right? The downside is that it looks longer and may be a turn-off for people who are just looking to switch over to Julia.

Reply:
The proposed syntax also gives you a `DataFrameRow`, so it depends on what you want to do exactly. You could show both.

Reply:
This is not the point. The proposed syntax is an equivalent of `df.loc['c']` in pandas. Unfortunately it is verbose, but it is only because `df.loc['c']` only appears to be simple and safe, and actually it is quite non-obvious what it does.

The equivalent of `df[df.id .== 'c', :]` in pandas is `df.iloc[df.index == 'c']`.

The fact that in this case the result was the same was only because `:id` has unique values. But in general it does not have to be this way (also pandas does not require key to be unique - and exactly as I have shown - in such case it does select the first value).

Reply:
Interesting observation that pandas isn't "type stable".  I changed the last index to 'e' (forcing a duplicate) and so this is how it works.
```
>>> df
   grp  x  y    z
a    1  6  4  3.0
b    2  5  5  4.0
c    1  4  6  5.0
d    2  3  7  6.0
e    1  2  8  7.0
e    2  1  9  NaN

>>> df.loc['c']
grp    1.0
x      4.0
y      6.0
z      5.0
Name: c, dtype: float64

>>> df.loc['e']
   grp  x  y    z
e    1  2  8  7.0
e    2  1  9  NaN
```

I agree that a DataFrameRow result matches better here.  Applying the suggestion...

Reply:
Hmm, so actually none of the two syntaxes would be an exact replacement, right? I still think showing both could be useful.